src/main/kotlin/pw/forst/tools/katlib/IterableExtensions.kt

```diff
@@ -497,3 +497,52 @@ inline fun <TItem> Iterable<TItem>.itemsToString(
     }
     return sb.toString().restrictLengthWithEllipsis(totalLength)
 }
+
+/**
+ * Transforms an Iterable into nullable elements while only keeping the non-null elements.
+ */
+fun <T, R> Iterable<T>.flatMapIndexedNotNull(function: (index: Int, T) -> Iterable<R>?): List<R> {
+    return flatMapIndexedTo(ArrayList(), function)
+}
+
+private inline fun <T, R, C : MutableCollection<in R>> Iterable<T>.flatMapIndexedTo(destination: C,
```

Comment:
I'd suggest making this function public as the other `mapTo` functions are public - thus allowing inlining of `flatMapIndexedNotNull`

Reply:
keep in mind, that all public functions must be documented

Reply:
Added documentation similar to the standard kotlin lib as the functions themselves are really pretty similar.